packages/vuetify/src/components/VSlider/__tests__/VSlider.spec.cy.tsx

```diff
@@ -193,4 +194,42 @@ describe('VSlider', () => {
       </VApp>
     ))
   })
+
+  // #16634
+  describe('rouding & decimals', () => {
+    it('should respect the decimals from both step and min', () => {
+      const currentValue = ref(1.001)
+
+      cy.mount(() => (
+        <VApp>
+          <CenteredGrid width="360px">
+            <VSlider
+              v-model={ currentValue.value }
+              step={1.001}
+              min={1.0001}
+              max={10}
+              thumb-label
+            />
+          </CenteredGrid>
+        </VApp>
+      ))
+
+      cy.get('.v-slider-thumb')
+        .trigger('mouseover')
+        .trigger('mousedown', { which: 1 })
+        .trigger('mousemove', 35, 0, { force: true }) // move to second step
+        .should(_ => {
+          expect(currentValue.value).equal(2.0011)
+        })
+        .trigger('mousemove', 190, 0, { force: true }) // move to fifth step
+        .should(_ => {
+          expect(currentValue.value).equal(6.0051)
+        })
+        .trigger('mousemove', 360, 0, { force: true }) // move to the final step
+        .should(_ => {
+          expect(currentValue.value).equal(10)
+        })
+        .trigger('mouseup')
+    })
```

Comment:
This is just a suggestion, but you could possibly rewrite this without the ref and v-model. Instead you can check `.emitted('.v-slider', 'update:modelValue').should('deep.equal', ...)` at the end.

Reply:
Thanks for the advice. It's much cleaner now.